### PR TITLE
Fix #5145: Improve treatment of soft modifiers

### LINKED ISSUE #5145: Inline as a soft keyword
You can currently define a variable whose name is `inline`, but you can't reference it
```scala
class Test {
  def foo(x: Int): Int = {
    val inline = 3
    inline
  }
}
```
```
-- [E103] Syntax Error: tests/allan/Test.scala:4:4 -----------------------------
4 |    inline
  |    ^^^^^^
  |    Illegal start of statement: no modifiers allowed here

longer explanation available when compiling with `-explain`
```

### PR DESCRIPTION
**Pull request by odersky**
- Generalize treatment of `inline` to a set of soft modifiers
 - Make `opaque` a soft modifier
 - Fix parsing of soft modifiers so that they are treated as modifiers
   only when preceding a hard modifier or a definition keyword

### CODE REVIEW

**Comment by odersky** [comment]
The syntaxHighlighting test fails because it stopped highlighting `inline` as a keyword in
```scala
class inline
```
The test should be fixed, but I don't know how to do it. @Duhemm can you help?

**Comment by Duhemm** [comment]
I think that @allanrenucci is the Syntax Highlighting Guru

File: compiler/src/dotty/tools/dotc/core/StdNames.scala [hunk 1]
```diff
@@ -436,6 +435,7 @@ object StdNames {
     val implicitConversions: N  = "implicitConversions"
     val implicitly: N           = "implicitly"
     val in: N                   = "in"
+    val `inline`: N             = "inline"
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/core/StdNames.scala]: Is this necessary with this change? Is it a style we want to recommend for soft keyword?

Comment (odersky) [reply compiler/src/dotty/tools/dotc/core/StdNames.scala]: It's no longer necessary. I removed the backticks.

**Comment by allanrenucci** [comment]
I have added `def foo(inline: Int) = 2` to the syntax highlighting tests

**Pull request merged**